tests/usecase/test_jdk_versions.py

```diff
@@ -0,0 +1,113 @@
+import basetest
+import buildpackutil
+import os
+import requests
+import time
+
+
+class TestJDKVersions(basetest.BaseTest):
+    def setUp(self):
+        super().setUp()
+
+    def _check_java_presence(self, target_dir):
+
+        try:
+            self.cmd(
+                (
+                    "cf",
+                    "ssh",
+                    self.app_name,
+                    "-c",
+                    os.path.join("/{}".format(target_dir), "bin", "java")
+                    + " -version",
+                )
+            )
+            rootCheckFail = False
+        except Exception:
+            rootCheckFail = True
+
+        try:
+            self.cmd(
+                (
+                    "cf",
+                    "ssh",
+                    self.app_name,
+                    "-c",
+                    os.path.join(
+                        "/home/vcap/app/.local/{}".format(target_dir),
+                        "bin",
+                        "java",
+                    )
+                    + " -version",
+                )
+            )
+            localCheckFail = False
+        except Exception:
+            localCheckFail = True
+
+        assert not (rootCheckFail and localCheckFail)
```

Comment:
I asked Matyas about this, because he's a bit more informed than I am:

> Generally I would go for the latter and make use of the dedicated assertions (eg. `assertListEqual`, `assertIsNone`, etc. - https://docs.python.org/3/library/unittest.html#assert-methods) that are available when using `unittest.TestCase` based test classes. Because _tests are specifications_, so readability and communicating intent is better (that’s why if there are more complex asserts, creating well-named custom assertion methods is good practice).
> 
> However if your test suite is written with the `pytest` style (eg. no test classes just methods), then you don’t have much choice but user plain `assert` statements. Although `pytest` is doing really good when dealing with these and reporting failures.
> 
> My personal preference is to stick to `unittest` features and keep `pytest` as a test-runner tool as much as possible. The reason is that even though `pytest` also provides everything you’d need, because of the plain `assert`s, readability suffers. The fixture functionality is extremely powerful, but that can also set back readability as sometimes you have to hunt around for fixture definitions as they are automagically injected, runtime, no imports or anything. (I actually had a hard time this morning when looking at an example cookiecutter project to find where a test fixture is coming from only to find out that it’s a `pytest` plugin with a single explicit occurrence in the requirements file, nowhere else).
> Furthermore you lock-in with `pytest` while in the future you might want to switch test-runners (parallelization, better reporting, etc.). (edited)

Reply:
So, tl;dr, please continue using the current unit test style, but transition to the `self.assert` methods.